### ISSUE TITLE
ProfileHTTP2 deployment fails with success response.

### ISSUE THREAD

**Issue by zongzw** [F5Networks/f5-appsvcs-extension#172]
When I did the following deployment, it was success, and created the HTTP2 Profile, but NOT be used. 

Whole AS3 declaration: 
```
{
  "class": "AS3",
  "action": "deploy",
  "persist": true,
  "declaration": {
        "class": "ADC",
        "schemaVersion": "3.0.0",
        "id": "43317603-6f80-4735-ab12-d12ad38e100a",
        "F5_610be7617fff469c88b71301cffd4c06": {
            "class": "Tenant",
            "label": "610be7617fff469c88b71301cffd4c06",
            "F5_7ed1ed90_cc91_43cc_8908_90174f7f9d5c": {
                "class": "Application",
                "label": "7ed1ed90-cc91-43cc-8908-90174f7f9d5c",
                "template": "generic",
                "F5_7aa368b8_2b05_46ac_b658_2c4063274a6f": {
                    "class": "Service_HTTP",
                    "label": "7aa368b8-2b05-46ac-b658-2c4063274a6f",
                    "pool": "F5_e5928b81_90e2_4f05_ae5b_39ee99a201e5",
                    "iRules": [
                        "choose_pool"
                    ],
                    "profileHTTP2": {
                    	"use": "F5_7d5054ff_8d15_45f6_947b_786375d419b5"
                    },
                    "virtualAddresses": [
                        "10.250.18.103"
                    ],
                    "virtualPort": 80,
                    "policyEndpoint": []
                },
                "choose_pool": {
                    "class": "iRule",
                    "remark": "choose private pool based on IP",
                    "iRule": "when CLIENT_ACCEPTED {}"
                },
                "F5_e5928b81_90e2_4f05_ae5b_39ee99a201e5": {
                    "class": "Pool",
                    "label": "e5928b81-90e2-4f05-ae5b-39ee99a201e5",
                    "loadBalancingMode": "round-robin",
                    "members": [
                        {
                            "servicePort": 80,
                            "serverAddresses": [
                                "10.250.17.111"
                            ],
                            "monitors": [
                                {
                                    "use": "F5_c16dd59b_ba9a_40e3_8b29_57ce26e7e297"
                                }
                            ]
                        }
                    ],
                    "monitors": [
                        {
                            "use": "F5_c5c19624_ab8a_41b2_9c53_01019677b811"
                        }
                    ]
                },
                "F5_c5c19624_ab8a_41b2_9c53_01019677b811": {
                    "class": "Monitor",
                    "label": "c5c19624-ab8a-41b2-9c53-01019677b811",
                    "monitorType": "http"
                },
                "F5_c16dd59b_ba9a_40e3_8b29_57ce26e7e297": {
                    "class": "Monitor",
                    "label": "c16dd59b-ba9a-40e3-8b29-57ce26e7e297",
                    "monitorType": "icmp"
                },
                "F5_7d5054ff_8d15_45f6_947b_786375d419b5": {
                "class": "HTTP2_Profile",
                "label": "7d5054ff-8d15-45f6-947b-786375d419b5",
                "activationMode": "alpn",
                "concurrentStreamsPerConnection": 10,
                "connectionIdleTimeout": 300,
                "enforceTlsRequirements": true,
                "frameSize": 2048,
                "headerTableSize": 4096,
                "insertHeaderName": "X-HTTP2",
                "receiveWindow": 32,
                "writeSize": 16384
            }
            }
        }
    }
}
```

AS3 responsed with 
```
{
    "results": [
        {
            "code": 200,
            "message": "success",
            "lineCount": 21,
            "host": "localhost",
            "tenant": "F5_610be7617fff469c88b71301cffd4c06",
            "runTime": 1123
        }
    ],
    "declaration": {
        "F5_610be7617fff469c88b71301cffd4c06": {
```

In fact, the HTTP2 Profile is not used. 
![图片](https://user-images.githubusercontent.com/17761849/66977486-bada9480-f0d8-11e9-9037-835a72f4b856.png)

Seen from BIGIP UI, if I manually select it, it says:

```01070734:3: Configuration error: In Virtual Server (/F5_610be7617fff469c88b71301cffd4c06/F5_7ed1ed90_cc91_43cc_8908_90174f7f9d5c/F5_7aa368b8_2b05_46ac_b658_2c4063274a6f) http2 specified activation mode requires a client ssl profile```

Should AS3 catch the error reason, and response in `results`?

**Comment by dstokesf5** [comment]
Yes, AS3 should be catching and reporting the error. After some experimentation with this, it looks like HTTP2 profile only really works with the Service_HTTPS class. The Service_HTTP class will not allow you to set the require ssl profile.

Are you able to switch to Service_HTTPS for your use case?

**Comment by zongzw** [comment]
@dstokesf5 Yes, we can switch the type to Service_HTTPS. In fact we have no dense requirement of setting HTTP2profile in Service_HTTP. 
Just reported the case.

**Comment by dstokesf5** [comment]
Thank you for reporting the issue. I just wanted to confirm that there was not a Service_HTTP use case I was overlooking. I have copied the bug into our internal product backlog as AUTOTOOL-864.

**Comment by walterk82** [comment]
Just for clarification, I believe that HTTP2 spec requires HTTPS only.

**Comment by dstokesf5** [comment]
This issue has been resolved as part of the 3.25.0 release.
